extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -346,7 +350,7 @@ export class AmpConsent extends AMP.BaseElement {
           CONSENT_ITEM_STATE.REJECTED,
           consentString);
     } else if (action == ACTION_TYPE.DISMISS) {
-      // dismiss
+      // TODO: Throw user warning at dismiss and defined consent string
```

Comment:
How come we need to throw a user warning here? 🤔 

If we already have a response, and they dismiss, shouldn't we just assume the previous response? My apologies if you had explained this previously.

Reply:
Good question, I'm glad that you ask : )

What we have today:  
1. Use the previously stored state. 
2. Still update/store the consent string with user dismiss. 

But @lannka and I agreed that we should keep the dismiss action same as before. That if action=dismiss, nothing will be stored. That's why I want to throw a user warning here. 

Dismiss behavior from this PR:
1. previous consent state will be applied. If there's none the state is UNKNOWN.
2. current consent string will be passed to vendors, but will not be stored. (I am still debating if we should simply discard the consent string if user dismiss. Please let me know if what do you think).

Reply:
Discussed offline:

This is actually a super interesting and hard to solve problem! It boils down to:

**What if we had a consent string and state, but then the Iframe passes back an invalid action?**

Which leads to:

* Ignore it?
* Store the valid state, but not the invalid string or vice versa
* Delete everything and go back to square 1?
* Just assume reject?

So, since we shouldn't make this important decision in a PR review, and this is what we are certain about in this case:

* Throw a user (site builders, publishers, 3p stakeholders, etc...) error on the AMP side (That way publishers and CMPs will see a clear error in their testing and environments).
* ???

We will be reaching out for more info on how to proceed with the new state.

@zhouyx for purposes of this PR, let's just change the warning to an error, and then leave a `TODO` so we can address this in another PR.

Reply:
Created #20010 to track the issue, also updated TODO. But I didn't include the error reporting in the PR sounds a bit unrelated to what this PR is trying to accomplish.